src/pages/Facility/settings/locations/components/LocationCard.tsx

```diff
@@ -80,16 +116,57 @@ export function LocationCard({ location, onEdit, className }: Props) {
         </div>
 
         <div className="mt-auto border-t border-gray-100 bg-gray-50 p-4">
-          <div className="flex justify-end">
-            <Button variant="outline" asChild>
-              <Link
-                href={`/location/${location.id}`}
-                className="flex items-center gap-2"
-              >
-                {t("view_details")}
-                <ChevronRight className="h-4 w-4" />
-              </Link>
-            </Button>
+          <div className="flex justify-between">
+            {!location.has_children && !location.current_encounter && (
+              <div>
+                <AlertDialog>
+                  <AlertDialogTrigger asChild>
+                    <Button variant={"white"}>
+                      <CareIcon icon="l-trash" />
+                    </Button>
+                  </AlertDialogTrigger>
+                  <AlertDialogContent>
+                    <AlertDialogHeader>
+                      <AlertDialogTitle>
+                        {t("remove")} {location.name}
```

Comment:
let's not do word by word translations; placement of words in a sentence might differ on different languages and would end up reading incorrect;

pass it as a variable and translate